### PR TITLE
VPA fix for handling empty resources section in container spec.

### LINKED ISSUE #811: VPA: jsonpatch add operation does not apply
I have just been starting to test the VPA (currently I am mostly interested in the "initial" to get request and limit resources set based on some metrics). So I created a statefulset without anything in the resource section and getting the following error:
```
  Warning  FailedCreate      3m (x6 over 6m)   statefulset-controller  create Pod api-proxy-redis-master-0 in StatefulSet api-proxy-redis-master failed error: Internal error occurred: Internal error occurred: jsonpatch add operation does not apply: doc is missing path: /spec/containers/0/resources/requests/memory
  Warning  FailedCreate      1m (x20 over 6m)  statefulset-controller  create Pod api-proxy-redis-master-0 in StatefulSet api-proxy-redis-master failed error: Internal error occurred: Internal error occurred: jsonpatch add operation does not apply: doc is missing path: /spec/containers/0/resources/requests/cpu
```
I would have expected the VPA to be able to set these fields even if they did not exist before.

Once I updated the `Statefulset` it all worked fine.

### PR DESCRIPTION
**Pull request by schylek**
fixes #811

### CODE REVIEW

File: vertical-pod-autoscaler/admission-controller/logic/server.go [hunk 1]
```diff
@@ -64,23 +64,47 @@ func (s *AdmissionServer) getPatchesForPodResourceRequest(raw []byte, namespace
 	patches := []patchRecord{}
 	updatesAnnotation := []string{}
 	for i, containerResources := range containersResources {
```
Comment (bskiba) [inline vertical-pod-autoscaler/admission-controller/logic/server.go]: Will this mean that we set the objects for all pods or only those controlled by VPA?

File: vertical-pod-autoscaler/admission-controller/logic/server.go [hunk 2]
```diff
@@ -64,23 +64,47 @@ func (s *AdmissionServer) getPatchesForPodResourceRequest(raw []byte, namespace
...
 			limit, found := containerResources.Limits[v1.ResourceMemory]
 			if found {
+				// Add limits map if missing
 				if pod.Spec.Containers[i].Resources.Limits == nil {
 					patches = append(patches, patchRecord{
 						Op:    "add",
 						Path:  fmt.Sprintf("/spec/containers/%d/resources/limits", i),
 						Value: v1.ResourceList{}})
 				}
+				// Set limt
```
Comment (bskiba) [inline vertical-pod-autoscaler/admission-controller/logic/server.go]: nit s/limt/limit/

Comment (schylek) [reply vertical-pod-autoscaler/admission-controller/logic/server.go]: Done.

File: vertical-pod-autoscaler/admission-controller/logic/server.go [hunk 3]
```diff
@@ -64,23 +64,47 @@ func (s *AdmissionServer) getPatchesForPodResourceRequest(raw []byte, namespace
 	patches := []patchRecord{}
 	updatesAnnotation := []string{}
 	for i, containerResources := range containersResources {
+
+		// Add resources object if missing
```
[the remark this answers is not in the document]
Comment (schylek) [reply vertical-pod-autoscaler/admission-controller/logic/server.go]: Done.

File: vertical-pod-autoscaler/admission-controller/logic/server.go (hunk 1 above)
Comment (schylek) [reply vertical-pod-autoscaler/admission-controller/logic/server.go]: Only for containers those controlled by VPA. Note that 'containersResources' contains recommendations - i.e. we only iterate if there are recommended values.

**Pull request merged**